apps/backend/src/modules/project-groups/project-group.repository.ts

```diff
@@ -122,6 +122,21 @@ class ProjectGroupRepository implements Repository {
 		};
 	}
 
+	public async findAllByUserId(userId: number): Promise<ProjectGroupEntity[]> {
+		const results = await this.projectGroupModel
+			.query()
+			.orderBy("createdAt", SortType.DESCENDING)
+			.withGraphJoined("[projects, users, permissions]")
+			.where("users.id", userId);
+
+		return results.map((projectGroup) =>
+			ProjectGroupEntity.initialize({
+				...projectGroup,
+				projectId: { id: (projectGroup.projects.pop() as ProjectModel).id },
```

Comment:
I'm not sure, but pay attention, that this pop mutates `projectGroup.projects` which was spreaded, so resulted object may loose some data. Also I think we should rename `projectId` field of `ProjectGroupEntity` to `project`, as having `projectId.id` is not good

Reply:
Here is the issue I'm talking about
![image](https://github.com/user-attachments/assets/edfc2cc6-7abd-49e6-ae64-6bbf24a47e18)
`toBeInitialized` should contain all three projects, not two

Reply:
Well i dont really get it, because i have correct amount of projects...

Reply:
agree with the @GvoFor comment, but firstly I don't understand why do we have projects array inside project-group if the project group is linked to only one project

Reply:
>firstly I don't understand why do we have projects array inside project-group if the project group is linked to only one project

Good question

Reply:
We don't need projects here, so removing the last and only project is completely safe. We are using an array because the database design includes a projects_to_project_groups table, as mentioned in the README. The migration for this was created a long time ago, establishing a many-to-many relationship between projects and their respective groups.

I changed the logic a little, it should be a better. However, changing the array of project Ids in an entity creates a mismatch with the database.